aspnetcore/signalr/authn-and-authz/sample/Hubs/ChatHub.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.SignalR;
+
+namespace SignalRAuthenticationSample.Hubs
+{
+    [Authorize]
+    public class ChatHub: Hub
+    {
+        public override async Task OnConnectedAsync()
+        {
+            await Clients.All.SendAsync("ReceiveSystemMessage", $"{Context.UserIdentifier} joined.");
+            await base.OnConnectedAsync();
+        }
+
+        public override async Task OnDisconnectedAsync(Exception exception)
+        {
+            await Clients.All.SendAsync("ReceiveSystemMessage", $"{Context.UserIdentifier} left.");
+            await base.OnDisconnectedAsync(exception);
+        }
+
+        public async Task SendToUser(string user, string message)
+        {
+            await Clients.User(user).SendAsync("ReceiveDirectMessage", $"{Context.UserIdentifier}: {message}");
```

Comment:
super nit: return

Reply:
I actually don't agree for samples. For our code, performance is crucial, but for samples I think it's easier if the user can simply add their own code around our sample without worrying about having to switch from `return`ing a task to `await`ing` it.

Reply:
😢 but lots of users don't know there is a difference and we're teaching them bad practices...
Although, probably better to accidentally allocate a state machine than to not await a `SendAsync` which is more common.

Reply:
My preference is to use best practices, unless it's too complicated. If we go with return task we need a side bar of  `return a task vs. await`

This sample returns a task. A production app might gain perf benefits from async.

The [intro into Razor Pages](https://docs.microsoft.com/en-us/aspnet/core/tutorials/razor-pages/page?view=aspnetcore-2.1) is all async